python/pyarrow/_csv.pyx

```diff
@@ -763,3 +765,86 @@ def open_csv(input_file, read_options=None, parse_options=None,
                  move(c_convert_options),
                  maybe_unbox_memory_pool(memory_pool))
     return reader
+
+
+cdef class WriteOptions(_Weakrefable):
+    """
+    Options for writing CSV files.
+
+    Parameters
+    ----------
+    include_header : bool, optional (default True)
+        Whether to include the header
+    batch_size : int, optional (default 1024)
+        How many rows to process together when converting and writing
+        CSV
+    """
+    cdef:
+        CCSVWriteOptions options
+
+    # Avoid mistakingly creating attributes
+    __slots__ = ()
+
+    def __init__(self, *, include_header=None, batch_size=None):
+        self.options = CCSVWriteOptions.Defaults()
+        if include_header is not None:
+            self.options.include_header = include_header
+        if batch_size is not None:
+            self.options.batch_size = 1024
+
+
+cdef _get_write_options(WriteOptions write_options, CCSVWriteOptions* out):
+    if write_options is None:
+        out[0] = CCSVWriteOptions.Defaults()
+    else:
+        out[0] = write_options.options
+
+
+def write_csv(data, output_file, write_options=None,
+              MemoryPool memory_pool=None):
+    """
+
+    Parameters
+    ----------
+    data: The data to write.
+        Either a pyarrow.RecordBatch or a pyarrow.Table
+    output_file: string, path, pyarrow.OutputStream or file-like object
+        The location of CSV data.
+    write_options: pyarrow.csv.WriteOptions
+        Options to configure writing the CSV file.
+    memory_pool: MemoryPool, optional
+        Pool for temporary allocations.
+
+    Returns
+    -------
+    None
+    """
+    cdef:
+        shared_ptr[COutputStream] stream
+        CCSVWriteOptions c_write_options
+        CMemoryPool* c_memory_pool
+        CRecordBatch* batch
+        CTable* table
+    _get_write_options(write_options, &c_write_options)
+
+    try:
+        where = _stringify_path(output_file)
+    except TypeError:
+        get_writer(output_file, &stream)
+    else:
+        c_where = tobytes(where)
+        stream = GetResultValue(FileOutputStream.Open(c_where))
```

Comment:
It seems `get_writer` should do this for string and path-like inputs already. Is there a reason you had to write this `try/except/else` switch?